packages/dom-helper/lib/prop.js

```diff
@@ -13,11 +16,26 @@ export function normalizeProperty(element, attrName) {
     // TODO should this be an o_create kind of thing?
     cache = {};
     for (key in element) {
-      cache[key.toLowerCase()] = key;
+      if (isSettable(element, key)) {
+        cache[key.toLowerCase()] = key;
+      } else {
+        cache[key] = UNDEFINED;
+      }
     }
     propertyCaches[tagName] = cache;
   }
 
   // presumes that the attrName has been lowercased.
-  return cache[attrName];
+  var value = cache[attrName];
+  return value === UNDEFINED ? undefined : value;
+}
+
+function isSettable(element, attrName) {
+  var desc = getOwnPropertyDescriptor(element.constructor.prototype, attrName);
+  if (!desc) { return true; }
+  if (!desc.writable || !desc.hasOwnProperty('value') && typeof desc.set !== 'function') {
```

Comment:
This is the confusing and important part. We can change the "value" of a property (e.g. what we would get back if we accessed it) if it is *writable OR it doesn't have a `value` AND has a `set` function*. Browser spec guarantees that we will never get back a descriptor that has both a `value` and a `set`. Always mutually exclusive.